bcolz/ctable.py

```diff
@@ -456,13 +455,14 @@ def addcol(self, newcol, name=None, pos=None, **kwargs):
         if len(newcol) != self.len:
             raise ValueError("`newcol` must have the same length than ctable")
 
-        if isinstance(newcol, np.ndarray):
-            if 'cparams' not in kwargs:
-                kwargs['cparams'] = self.cparams
+        if self.rootdir is not None:
```

Comment:
I agree, adding columns to an existing on-disk table should make the carray also on-disk unless otherwise specified.